File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int maxDepth(TreeNode root) {
        return maxDepthHelper(root, 0);
    }

    public int maxDepthHelper(TreeNode root, int h) {
        if(root == null)    {
            return h;
        }

        int l = maxDepthHelper(root.left, h + 1);
        int r = maxDepthHelper(root.right, h + 1);

        return Math.max(l, r);
    }
}

File: Data Structures & Algorithms/binary-tree-maximum-path-sum/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxPathSum(self, root: Optional[TreeNode], currSum=0) -> int:
        self.best = -1001
        
        def dfs(root):
            if not root:
                return 0
            
            left_gain = max(0, dfs(root.left))
            right_gain = max(0, dfs(root.right))

            curr_path = root.val + left_gain + right_gain
            self.best = max(self.best, curr_path)

            return root.val + max(left_gain, right_gain)

        dfs(root)
        return self.best

